### PR TITLE
fix[tool]: fix output formats for .vyz files

### PR DESCRIPTION
**Pull request by charles-cooper**
### Commit message

```
run the format name through the translate map. for instance,
`annotated_ast` output format was not working for `.vyz` files.

this commit has some additional fixes that were discovered when adding
the integration tests and refactoring related to the settings not
getting propagated uniformly across different entry points.
```

### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by harkal — APPROVED** [review]

**Review by fubuloubu — APPROVED** [review]

**Comment by cyberthirst** [comment]
```python
def test_archive_compile_options(input_files):
    tmpdir, _, _, contract_file = input_files
    search_paths = [".", tmpdir]

    options = ["abi_python", "json", "ast", "annotated_ast", "ir_json"]

    for option in options:
        out = compile_files([contract_file], ["archive_b64", option], paths=search_paths)

        archive_b64 = out[contract_file].pop("archive_b64")

        archive_path = Path("foo.zip.b64")
        with archive_path.open("w") as f:
            f.write(archive_b64)

        # compare compiling the two input bundles
        out2 = compile_files([archive_path], [option])

        if option in ["ast", "annotated_ast"]:
            # would have to normalize paths and imports, so just verify it compiles
            continue

        assert out[contract_file] == out2[archive_path]
  ```

File: tests/unit/cli/vyper_compile/test_compile_files.py [hunk 1]
```diff
@@ -360,6 +360,104 @@ def test_archive_b64_output(input_files):
...
+        # compare compiling the two input bundles
+        out = compile_files([contract_file], [option], paths=search_paths)
+        out2 = compile_files([archive_path], [option])
+
+        if option in ["ast", "annotated_ast"]:
+            # would have to normalize paths and imports, so just verify it compiles
+            continue
+
+        if option in ["ir_runtime", "ir", "asm", "archive", "solc_json"]:
+            # TODO investigate why these don't pass the assert
```
Comment (cyberthirst) [inline tests/unit/cli/vyper_compile/test_compile_files.py:440]: the should be correct, but would like to explore this before approving

Comment (charles-cooper) [reply tests/unit/cli/vyper_compile/test_compile_files.py:440]: i suspect that has to do with codegen.reset_names(), which was fixed in 8f433f8de9ec3ead39e1691c45e2821fe8e3922b

Comment (charles-cooper) [reply tests/unit/cli/vyper_compile/test_compile_files.py:440]: cf. as well: `vyper/ir/compile_ir.py:_next_symbol` may need to be reset.

Comment (charles-cooper) [reply tests/unit/cli/vyper_compile/test_compile_files.py:440]: the assembly one should be fixed here: https://github.com/vyperlang/vyper/pull/4338/commits/e09f589607046cb16be9b09ff50ab984d8e79bc7

the others might need a bit more investigation. but i think the important thing here is that the bytecode output and integrity hashes are the same; if some metadata is different that is not such a big deal.

Comment (cyberthirst) [reply tests/unit/cli/vyper_compile/test_compile_files.py:440]: debugged this a bit further and got the following:
```python
def _eq_helper(ir1, ir2):
    assert isinstance(ir1, IRnode) and isinstance(ir2, IRnode)
    if isinstance(ir1, IRnode):
        for n1, n2 in zip(ir1.args, ir2.args):
            if n1 != n2:
                if isinstance(n1, IRnode) and n1.value == "external_contract":
                    print("value comparison:", n1.value == n2.value)
                    print("args comparison:", n1.args == n2.args)
                    print("typ comparison:", n1.typ == n2.typ)
                    print("location comparison:", n1.location == n2.location)
                    print("mutable comparison:", n1.mutable == n2.mutable)
                    print("add_gas_estimate comparison:", n1.add_gas_estimate == n2.add_gas_estimate)
            if isinstance(n1, IRnode):
                _eq_helper(n1, n2

_eq_helper(out[contract_file]['ir_runtime'], out2[archive_path]['ir_runtime'])
```
output:
```
value comparison: True
args comparison: True
typ comparison: False
location comparison: True
mutable comparison: True
add_gas_estimate comparison: True
```

typ is `InterfaceT` where we don't have `def __eq__(self, other):` method but inherit from `_UserType` which defines:
```python
    def __eq__(self, other):
        return self is other
```

**Review by cyberthirst — APPROVED** [review]

File: tests/functional/venom/test_venom_repr.py [hunk 2]
```diff
@@ -45,11 +51,17 @@ def _loop() -> uint256:
...
+def test_round_trip_sources(vyper_source, debug, optimize, compiler_settings, request):
     """
     Test vyper_sources round trip
     """
     vyper_source = textwrap.dedent(vyper_source)
+
+    if debug and optimize == OptimizationLevel.CODESIZE:
+        # FIXME: some round-trips fail when debug is enabled due to labels
+        # not getting pinned
+        request.node.add_marker(pytest.mark.xfail(strict=False))
```
Comment (charles-cooper) [inline tests/functional/venom/test_venom_repr.py:63]: to extract this to a new issue

**Pull request merged**